backend/fms_core/template_importer/sheet_data.py

```diff
@@ -40,25 +40,33 @@ def prepare_rows(self):
         for row_id in data_row_ids_range(self.header_row_nb + 1, self.dataframe):
             row_data = self.dataframe.iloc[row_id]
             self.rows.append(row_data)
-
-            row_repr = f"#{row_id + 1}"
+            row = row_id + 1
+            row_repr = f"#{row}"
+            row_str_data = panda_values_to_str_list(row_data)
 
             result = {
                 'row_repr': row_repr,
-                'diff': [row_repr] + panda_values_to_str_list(row_data),
+                'diff': [row_repr] + row_str_data,
                 'errors': [],
                 'validation_error': ValidationError([]),
                 'warnings': [],
             }
             self.rows_results.append(result)
+            
+            #checks if entire row has missing data (empty cells)
+            self.empty_row = row if not any(row_str_data) else self.empty_row
 
+        #checks if empty row exists in sheet_data, appends error with the last filled line in the sheet_data
+        if self.empty_row:
+            erroneous_row = self.empty_row + 1
+            self.base_errors.append(f'Empty line detected. Fill in empty line at row #{str(self.empty_row)} or remove data at row #{str(erroneous_row)}.')
```

Comment:
Given that you might have multiple empty lines you might tweak the message to match possible outcomes ... Maybe "... Fill in empty lines up to row ..."